### ISSUE TITLE
Gitblit is live

### ISSUE THREAD

**Issue by smeligrana** [gitblit/gitblit#1300]
Hello,
    but is the gitblit project still alive?

Thanks

**Comment by rnveach** [comment]
similar to https://github.com/gitblit/gitblit/issues/1265 and https://github.com/gitblit/gitblit/issues/1274

**Comment by fzs** [comment]
Yes, because it has not been given up. I have a vital interest in it.
No, because right now no one finds the time to release an update.
I would like to release a new version this year, but cannot promise
anything.

**Comment by carstenartur** [comment]
Great to hear that there is still somebody taking care. I could make use of it but the list of changes I would like to see seems to be long.
Eg. for the junit tests it is accessing a non existing github repository at
https://github.com/git/hello-world.git
Played around with it at https://github.com/carstenartur/gitblit but still no real success.

**Comment by fzs** [comment]
One of my main concerns is regression testing and testability. As Carsten
pointed out, for unit tests it currently makes use of externally hosted
repositories. Which are gone by now.
I think that we need are one or more prepared repositories checked in, e.g.
as a tarball, which can be unpacked for unit tests. These should then cover
all sorts of test cases, and can be changed and simply be unpacked again
for the next test, to get a clean state.

This is of course not the most interesting task, so no one has tackled it
yet. But it is necessary to get better test coverage for me to be able to
merge patches of code that I don‘t know well.
I mentioned that in #1265, but nothing happened so far.

**Comment by carstenartur** [comment]
I created a github repository for the tests and changed the code to use it but it does not look like it is really completely working. So I guess there is a kind of compatibility problem with github and gitblit/jgit now. You can see what I tested in my fork. In the end I think it is a good idea to have a test checking compatibility as an integration test. However for all other tests we should use the jgit in memory git repository. It is fast and should be suitable for nearly all cases where you need an externel repository.

**Comment by rnveach** [comment]
A while ago, I started an example of adding a test for some servlets and wickets for using in-memory created repositories as an example of how I think tests should go for the repository. Seems similar to @carstenartur but I couldn't find his new tests in his repository.

See https://github.com/rnveach/gitblit/commit/d4b7fecc592f4ca80f0ba99fbbf019907dea7bac#diff-40dc38b8e1cc8224c3b9602657444c93 and https://github.com/rnveach/gitblit/commit/d2a60145a7d56cef127ac1369f2bfca68f88be2b#diff-498dd45fb46cf0c53e2a8da9454edb0b

@fzs If this is the way you want to go, I can start a PR bringing the basics of what I have in and anyone can start extending it.

**Comment by carstenartur** [comment]
@rnveach I did not yet create new tests using the described approach with a in memory repository using jgit in gitblit. I did it in another project. In the gitblit fork I just changed the reference to the github repository to a newly created one.
see https://github.com/carstenartur/gitblit/commit/0a54e613f45339f9de06682dc4a380a7d6878bf7
https://github.com/carstenartur/hello-world.git
But as I said I must have made something wrong or there is a incompatibility, not sure about this.

**Comment by smeligrana** [comment]
@carstenartur to solve the problem of the repository (helloworld) I have created one with jgit in the tmp folder. https://github.com/smeligrana/gitblit

**Comment by wirmachenbunt** [comment]
bump
a new release would be totally cool. especially if the collapsible group feature gets merged

**Comment by chirontt** [comment]
I've tried to use carstenartur's and smeligrana's hello-world.git repos but they don't pass the tests, due to the facts that many tests require specific things present in that repo, such as it should contain a "java.java" file with at least 2 commits to it, or the number of commits since 2008-07-15 is exactly 12, etc.

I've now re-created the hello-world.git repo fitting the above test requirements. I've checked it into my fork at @chirontt/gitblit, to the src/test/data folder. Please have a go at it. As of now, the failures in GitBlitSuite test suite in my fork are now reduced to just 2 classes: LuceneExecutorTest and RedisTicketServiceTest.

RedisTicketServiceTest can't be fixed because it requires an external Redis service. Maybe it could be fixed with a mock Redis service? or a Redis service simulation (like @SpectoLabs/hoverfly)?

LuceneExecutorTest need someone with better knowledge of Lucene to work out why it fails, as I'm baffled by it.

**Comment by wirmachenbunt** [comment]
@chirontt , can you make a release / compile with all the changes since 2016 ?

**Comment by chirontt** [comment]
A release has to be done by a member of the gitblit team, not by me.

@wirmachenbunt, you can build it yourself from the master source at @gitblit/gitblit which should include all changes committed since the last release (1.8.0, in June 2016).

Assuming you have the following commands available in your path: git, ant, javac (from the JDK), then you can just clone the official gitblit repository to your local drive at the command-line:

_git clone http://github.com/gitblit/gitblit.git_

You would have a gitblit subfolder containing the complete source code of gitblit. Go into that gitblit folder and run:

_ant buildGO_

to build the gitblitGO server program, available in the build/target/gitblit-1.9.0-SNAPSHOT.zip (for Windows) or build/target/gitblit-1.9.0-SNAPSHOT.tar.gz (for Linux)

If you want to produce just the gitblit.war file, the build command is:

_ant buildWAR_

and the war file would be produced in build/target/gitblit-1.9.0-SNAPSHOT.war

**Comment by wirmachenbunt** [comment]
@chirontt 
thanks for the hint, i tried building it but get the build error "error: package javax.activation does not exist" 
any idea how to solve this. i use JDK 7.0

**Comment by chirontt** [comment]
@wirmachenbunt, I don't have such problem in compiling the gitblit source,
either with JDK 7 or 8, although it has errors generating the docs (i.e.
converting *.mkd files to *.html files.)

Here are my JDK 7 and Ant info:

D:\dev\gitblit>ant -version
Apache Ant(TM) version 1.9.8 compiled on December 25 2016

D:\dev\gitblit>java -version
java version "1.7.0_80"
Java(TM) SE Runtime Environment (build 1.7.0_80-b15)
Java HotSpot(TM) Client VM (build 24.80-b11, mixed mode, sharing)

In fact gitblit doesn't use the activation-1.1.jar anyway which contains
the javax.activation package.
